types/priv_validator/socket.go

```diff
@@ -34,35 +37,65 @@ const (
 	dialRetryIntervalSeconds = 1
 )
 
-// NewPrivValidatorSocket returns an instance of PrivValidatorSocket.
-func NewPrivValidatorSocketClient(logger log.Logger, socketAddr string) *PrivValidatorSocketClient {
+// NewPrivValidatorSocketClient returns an instance of
+// PrivValidatorSocketClient.
+func NewPrivValidatorSocketClient(
+	logger log.Logger,
+	socketAddr string,
+	privKey *crypto.PrivKeyEd25519,
+) *PrivValidatorSocketClient {
 	pvsc := &PrivValidatorSocketClient{
 		SocketAddress: socketAddr,
+		privKey:       privKey,
 	}
+
 	pvsc.BaseService = *cmn.NewBaseService(logger, "privValidatorSocketClient", pvsc)
+
 	return pvsc
 }
 
+// OnStart implements cmn.Service.
 func (pvsc *PrivValidatorSocketClient) OnStart() error {
 	if err := pvsc.BaseService.OnStart(); err != nil {
 		return err
 	}
 
-	var err error
-	var conn net.Conn
+	var (
+		err  error
+		conn net.Conn
+	)
+
 RETRY_LOOP:
 	for {
 		conn, err = cmn.Connect(pvsc.SocketAddress)
 		if err != nil {
-			pvsc.Logger.Error(fmt.Sprintf("PrivValidatorSocket failed to connect to %v.  Retrying...", pvsc.SocketAddress))
+			pvsc.Logger.Error(
+				"OnStart",
```

Comment:
I think the error msg should be along the lines "Failed to connect to server"

Reply:
See my [comment](https://github.com/tendermint/tendermint/pull/1204/files#r167945574) for some clarification that'd be important for me. :>